frontend/src/stores/useSelectionStore.ts

```diff
@@ -17,17 +30,17 @@ const useSelectionStore = create(set => ({
       .getState()
       .project
       ?.states
-      ?.map(s => s.id) ?? [],
+      ?.map((s: AutomataState) => s.id) ?? [],
     selectedTransitions: useProjectStore
       .getState()
       .project
       ?.transitions
-      ?.map(t => t.id) ?? [],
+      ?.map((t: AutomataTransition) => t.id) ?? [],
     selectedComments: useProjectStore
       .getState()
       .project
       ?.comments
-      ?.map(c => c.id) ?? []
+      ?.map((c: ProjectComment) => c.id) ?? []
```

Comment:
```suggestion
      ?.map(s => s.id) ?? [],
    selectedTransitions: useProjectStore
      .getState()
      .project
      ?.transitions
      ?.map(t => t.id) ?? [],
    selectedComments: useProjectStore
      .getState()
      .project
      ?.comments
      ?.map(c => c.id) ?? []
```
`s`, `t`, and `c` are inferred since `map` knows the type of the array its operating on

Reply:
Hmm okay IDE was telling me they were any so I thought it was a weird case where they weren't being inferred but I now realise that that's because of the issues with the imports/hinting. I realise at this point i should really just not trust the IDE lol but will double check with the imports next time

Reply:
After pulling I have gotten the fix for the import errors yay 🎉 So shouldn't be getting misled by that again (will also just distrust IDE from now on ofc)